File: 2차 과제/3-4/c 소스 파일/3-4.c
#include <stdio.h>
#include "3-1-1.c"

void print(int* list, int len, int count);
int insert(int* list, int len);
int delete(int* list, int len);

int main() {
    int list[8]={10, 20, 40, 50, 60, 70};
    int len=0;
    int count=0;

    //초반 원소 개수
    for (int i = 0; list[i]!=0; i++) len++;
    print(list, len, count);

    //삽입
    count=insert(list, len);
    // 삽입 후 선형 리스트 길이
    len=0;
    for (int i = 0; list[i]!=0; i++) len++;
    print(list, len, count);

    // 삭제
    count=delete(list, len);
    // 삭제 후 선형 리스트 길이
    len=0;
    for (int i = 0; list[i]!=0; i++) len++;
    print(list, len, count);
    
    return 0;
}

File: 2차 과제/3-4/py 소스파일/3-4.py
from func_list3_4 import print_def
from func_list3_4 import insert
from func_list3_4 import delete

def main() :
    arr = [10, 20, 40, 50, 60, 70, 0, 0]
    count=0
    size=0

    # 초반 형태 및 개수
    for i in range(8):
        if arr[i]!=0: size+=1
    print_def(arr, size, count)

    # 삽입 후
    count=insert(arr, size)
    size=0
    for i in range(8):
        if arr[i]!=0: size+=1
    print_def(arr, size, count)

    #삭제 후
    count=delete(arr, size)
    size=0
    for i in range(8):
        if arr[i]!=0: size+=1
    print_def(arr, size, count)    

if __name__ == "__main__":
    main()